Compatibility with Android Studio Meerkat | 2024.3.1 Canary 7 (2024.3.1.7)

raghavsatyadev opened the pull request:
- Updated the IDE version to 2024.3.1.6. -> gradle.properties
- added JVM Args and Kotlin Code Style. -> gradle.properties
- Updated Gradle Wrapper to 8.12-rc-2. -> gradle-wrapper.properties
- Updated Java compatibility to version 21. -> build.gradle.kts
- Updated notification groups to use the NotificationGroupManager API. -> NotificationHelper.kt
- Removed deprecation in ReflectKt.kt
- Added Gradle Script file to run "buildPlugin runIde" commands directly without creating a configuration manually. -> BuildRun.run.xml
- Updated .gitignore to exclude .kotlin folder

pbreault commented on src/main/kotlin/com/developerphil/adbidea/ui/NotificationHelper.kt:
```diff
@@ -1,20 +1,54 @@
+        )
 
-    fun error(message: String) = sendNotification(message, NotificationType.ERROR, ERRORS)
+        // Set the listener separately, if needed
+        notification.setListener(NotificationListener.URL_OPENING_LISTENER)
```
This method is deprecated. When does the listener need to be set separately?

raghavsatyadev replied:
![image](https://github.com/user-attachments/assets/90329120-dd0e-45de-8342-2d50c65cf7d6)

We have two options: provide a Notification Listener in createNotification() or set it separately using setListener(). 

createNotification() is scheduled for removal, but setListener() is still marked as deprecated.

However, from the look at the project, we don't need listeners as we are not providing any links in the Notification. I'm removing the listener from both usages. Please let me know if you'd like them to be there.

pbreault reviewed (commented):
Thanks for the pull request! Looks good to me in general but I have a few comments/questions that need to be addressed before merging it.

raghavsatyadev commented:
@pbreault Added the changes that you requested. Plus:

- Removed more deprecations
- Removed unused imports
- fixed some basic warnings from the IDE

pbreault commented on gradle.properties:
```diff
@@ -1,10 +1,14 @@
-sinceBuild=242.23339
+sinceBuild=243.22562.145
+
+# Use the latest of Android plugin version
+# see https://github.com/JetBrains/intellij-platform-gradle-plugin/releases
+androidPluginVersion=243.22562.218
```
This field isn't used, let's remove it.

pbreault commented on src/main/kotlin/com/developerphil/adbidea/adb/DeviceResultFetcher.kt:
```diff
@@ -113,6 +113,9 @@ sealed class DeviceResult {
+    data object Cancelled : DeviceResult()
+    data object DeviceNotFound : DeviceResult()
 }
+//
+//val Module.isAndroidApp: Boolean
+//    get() = AndroidFacet.getInstance(this)?.configuration?.isAppProject ?: false
```
If this method isn't use, we can remove it instead of commenting it out.

pbreault commented on src/main/resources/META-INF/plugin.xml:
```diff
@@ -31,108 +37,90 @@
+            </action>
+            <separator />
+
+            <action class="com.developerphil.adbidea.action.RevokePermissionsAction"
+                    description="Revoking the runtime permissions of the current application"
+                    id="com.developerphil.adbidea.action.RevokePermissionsAction"
+                    text="ADB Revoke Permissions" />
```
The change hasn't been reverted. The order should be id, class, text, description.

pbreault commented on src/main/resources/META-INF/pluginIcon.svg:
Please remove the icon.

pbreault reviewed (changes requested):
This PR is going beyond it's original purpose of bringing compatibility with the `2024.3.1`. This is making it hard to review. Can you break this down in smaller more focused Pull Requests?

raghavsatyadev commented:
@pbreault changes done as requested

raghavsatyadev commented:
@pbreault if you got time, can you please review this PR?

pbreault commented:
Sorry for the delay. Thanks for making the changes! Good to merge now 👍
